kafka-to-kv/kafka-to-kv.py

```diff
@@ -0,0 +1,109 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import json
+import base64
+
+import requests
+
+NGINX_HOST = os.environ['NGINX_HOST']
+NGINX_PORT = os.environ['NGINX_PORT']
+TABLE_NAME = os.environ['TABLE_NAME']
+CONTAINER_NAME = os.environ['CONTAINER_NAME']
+EVENT_KEY = os.environ['EVENT_KEY']
+USERNAME = os.environ['USERNAME']
+PASSWORD = os.environ['PASSWORD']
+
+def get_request_url():
+
+    return 'http://{0}:{1}/{2}/{3}/'.format(NGINX_HOST, NGINX_PORT, CONTAINER_NAME, TABLE_NAME)
+
+
+def create_encoded_auth():
+    s = '{0}:{1}'.format(USERNAME, PASSWORD)
+    b = bytearray()
+    b.extend(map(ord, s))
+    res = base64.encodestring(b)
+    res = res.decode('utf-8').replace('\n', '')
+    return res
+
+
+def get_request_headers(v3io_function):
+    encoded_auth = create_encoded_auth()
+    return {
+        'Content-Type': 'application/json',
+        'Cache-Control': 'no-cache',
+        'X-v3io-function': v3io_function,
+        'Authorization': 'Basic {0}'.format(encoded_auth),
+    }
+
+
+def send_request(payload, logger, url, headers):
+
+    try:
+        response = requests.post(url, json=payload, headers=headers, timeout=1)
+        logger.debug(response.status_code)
+        logger.debug(response.content)
+    except Exception as e:
+        logger.error("ERROR: {0}".format(str(e)))
+
+
+def insert_key(item,key):
+    item['Key'] = {"name": {"S": key}}
+    return item
+
+
+def insert_attributes(item, attributes, allow_null_attributes=True):
+    if attributes:
+        item['Item'] = {}
+        for attr, definition in attributes.items():
+            item['Item'][attr] = {}
+            for key, value in definition.items():
+                if allow_null_attributes or value is not None:
+                    item['Item'][attr][key] = value
+            if not item['Item'][attr]:
+                item['Item'].pop(attr)
+    return item
+
+
+def generate_attributes(json_object, object_key):
+   attributes = {}
+   for key in json_object:
+       if key != object_key:
+           if type(json_object[key]) == bool:
+               attributes[key] = {"BOOL": json_object[key]}
+           if type(json_object[key]) == int:
+               attributes[key] = {"N": str(json_object[key])}
+           if type(json_object[key]) == float:
+               attributes[key] = {"N": str(json_object[key])}
+           if type(json_object[key]) == str:
+               attributes[key] = {"S": json_object[key]}
+   return attributes
+
+
+def generate_payload(event_body):
+    msg = json.loads(event_body)
+    attributes = generate_attributes(msg, EVENT_KEY)
+    payload = {}
+    payload = insert_key(payload, msg[EVENT_KEY])
+    payload = insert_attributes(payload, attributes)
+    return payload
+
+
+def handler(context, event):
```

Comment:
Handler should be first function, followed by init_context (here you don't have it, but if you do that's the convention).

All other functions should start with an underscore